core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1993,6 +1993,12 @@ private void renameInternal(RpcContext rpcContext, LockedInodePath srcInodePath,
 
     // Now we remove srcInode from its parent and insert it into dstPath's parent
     renameInternal(rpcContext, srcInodePath, dstInodePath, false, context);
+
+    // Check options and determine if we should schedule async persist. This is helpful for compute
+    // frameworks that use rename as a commit operation.
+    if (context.getPersist() && srcInode.isFile() && !srcInode.isPersisted()) {
```

Comment:
Does this work if the file was already written with async persist?

Reply:
Yeah this works exactly the same as calling `scheduleAsyncPersist` on the file.